pkg/subscriber/github/git_subscriber_item.go

```diff
@@ -323,29 +328,30 @@ func (ghsi *SubscriberItem) subscribeResourceFile(hostkey types.NamespacedName,
 	pkgMap[dpltosync.GetName()] = true
 
 	klog.V(4).Info("Ready to register template:", hostkey, dpltosync, syncsource)
+	err = ghsi.synchronizer.AddTemplates(syncsource, hostkey, []kubesynchronizer.DplUnit{{Dpl: dpltosync, Gvk: *validgvk}})
 
-	err = ghsi.synchronizer.RegisterTemplate(hostkey, dpltosync, syncsource)
 	if err != nil {
 		err = utils.SetInClusterPackageStatus(&(ghsi.Subscription.Status), dpltosync.GetName(), err, nil)
 
+		ghsi.successful = false
+
 		if err != nil {
 			klog.V(4).Info("error in setting in cluster package status :", err)
 		}
 
-		pkgMap[dpltosync.GetName()] = true
-
 		return err
 	}
 
 	dplkey := types.NamespacedName{
 		Name:      dpltosync.Name,
 		Namespace: dpltosync.Namespace,
 	}
-	kvalid.AddValidResource(*validgvk, hostkey, dplkey)
 
 	pkgMap[dplkey.Name] = true
+	//this is for the adaption of the new synchronizer API
+	klog.V(10).Infof("ingnor this kvalid %v, only exist for adopting synchronizer API", kvalid)
```

Comment:
Minor: ingnor should be ignore?

Reply:
Hmmm, if we dont have  print statement like this, then we have to the `kvalid` within the function... 

However, with the new code changes, we don't actually need to use the `kvalid`. The idea way is change the function signature to get rid of the `kvalid` parameter... but I don't want to change the original call chain that much... that's why I did the above print..

Reply:
I was referring to your spelling mistake.

Reply:
sure